### PR TITLE
Enhancement: Method create_nested_dict_from_dots_in_keys uses stack instead of recursion

### PR DESCRIPTION
**Pull request by mithunbharadwaj**

### CODE REVIEW

File: asabiris/formatter/jinja/service.py [hunk 1]
```diff
@@ -37,21 +36,22 @@ def create_nested_dict_from_dots_in_keys(data):
...
-			parts = key.split('.')
-			current_dict = nested_dict
-			for part in parts[:-1]:
-				current_dict.setdefault(part, {})
-				current_dict = current_dict[part]
+	nested_dict = {}  # Initialize an empty nested dictionary
+	stack = [(nested_dict, key, value) for key, value in data.items()]  # Create a stack of (dictionary, key, value) tuples
 
-			current_dict[parts[-1]] = value
+	while stack:
```
Comment (eliska-n) [inline asabiris/formatter/jinja/service.py:44]: Cool. I should read more about this structure. However, the recursion is missing here, I am afraid. This dictionary is not processed right: 
```
{'alert': {'event.other.id': '1234'}}
```

**Review by eliska-n — CHANGES_REQUESTED** [review]

File: asabiris/formatter/jinja/service.py (hunk 1 above)
Comment (mithunbharadwaj) [reply asabiris/formatter/jinja/service.py:44]: @eliska-n I will test it more! Thanks Eliska

Comment (mithunbharadwaj) [reply asabiris/formatter/jinja/service.py:44]: @eliska-n I will see what I can do.

File: asabiris/formatter/jinja/service.py [hunk 2]
```diff
@@ -27,31 +26,54 @@ async def format(self, template_path, template_params):
...
-			current_dict = nested_dict
-			for part in parts[:-1]:
-				current_dict.setdefault(part, {})
-				current_dict = current_dict[part]
-
-			current_dict[parts[-1]] = value
-		else:
-			nested_dict[key] = value
-	return nested_dict
+	def create_nested_dict_from_dots_in_keys(self, data):
```
Comment (eliska-n) [inline asabiris/formatter/jinja/service.py:32]: it does not work for this dictionary:
```
	{
		"alert": {
			"event.id": 2
		},
		"happy.unicorn": {
			"best.friend": "Rabbit"
		}
	}
```
nor for this one:
```
	{
		"happy.unicorn": {
			"best.friends": {
				"rabbit": "Joe",
				"cat": "Ema",
				"praying.mantis": "Sisi"
			}
		}
	}
```
keys followed by dictionary as a value are not "parsed"

**Review by eliska-n — CHANGES_REQUESTED** [review]

**Review by ateska — CHANGES_REQUESTED** [review]
We need to have a unit test for this.

File: asabiris/formatter/jinja/service.py [hunk 3]
```diff
@@ -27,31 +26,48 @@ async def format(self, template_path, template_params):
 		template = jinja2.Template(template_io.read().decode('utf-8'))
 
 		# This creates nested dictionary from keys with dot notation
-		template_params = create_nested_dict_from_dots_in_keys(template_params)
+		template_params = self.create_nested_dict_from_dots_in_keys(template_params)
 		return template.render(**template_params)
 
+	def create_nested_dict_from_dots_in_keys(self, data):
+		"""
```
Comment (ateska) [inline asabiris/formatter/jinja/service.py:33]: See issue from @eliska-n above.

**Review by ateska — CHANGES_REQUESTED** [review]

File: asabiris/formatter/jinja/service.py (hunk 3 above)
Comment (mithunbharadwaj) [reply asabiris/formatter/jinja/service.py]: @ateska I will add unit test

**Review by ateska — APPROVED** [review]

**Comment by ateska** [comment]
Please resolve conflict and merge this if possible.

**Comment by eliska-n** [comment]
If the tests are passing, unicorns are happy :unicorn:

**Pull request merged**